### ISSUE TITLE
Does Catalyst use eval() during evaluation

### ISSUE THREAD

**Issue by MathieuHaller** [catalyst-team/catalyst#1323]
Hello,

I realize this is a very basic question, but I've gone through the docs + source code and haven't found the answer.  
Does catalyst automatically put the layers of the model in eval mode (like model.eval() would) when predicting on the validation set during training and when calling functions such as predict_loader on the validation/infer sets ?

Or maybe it's not needed as such functions deal with batches and so batchnorm layers work just fine as is, without eval mode ?

Thank you for your time.

**Comment by Scitator** [comment]
Long story short,
- "Yes" for the [predict_loader](https://github.com/catalyst-team/catalyst/blob/master/catalyst/runners/runner.py#L668)
- "No" for the [predict_batch](https://github.com/catalyst-team/catalyst/blob/master/catalyst/runners/runner.py#L518), cause you have to implement this logic on your own usually

in such a case
- you could easily predict any loader with the `predict_loader` method and eval mode
- you could write your own loader predictor with `predict_batch` under the hood

**Comment by MathieuHaller** [comment]
Alright, thank you for the clarification !
